### PR TITLE
WIP TASK: improve performance of initial state generation

### PR DESCRIPTION
**Pull request by dimaip**
Related: #1579 
Resolves: #1508

This PR does the following:
- reconstructs parent->child relations on the client for each change, for all nodes. But in-memory it's basically free, 0.3ms for 500 nodes, so no need to bother about that.
- loads default nodes for tree asynchronously. This is especially important, since now it's rather slow (due to policy enforcement)
- loads nodes for tree with `$omitMostPropertiesForTreeState`, which drastically reduces the payload size

### CODE REVIEW

**Comment by dimaip** [comment]
So here comes actual performance analyses.
I've been testing on a website with **540 document nodes**, expanding the tree to show all of them.

1. The `neosUiDefaultNodes` FQ operation uses recursive `getChildNodes` calls and takes `2.7s`. While that is not super fast, I don't know how to improve that.

2. Now for the `renderNode` rendering itself.

The biggest cost are our own policy aspects: https://github.com/neos/neos-ui/blob/master/Classes/Neos/Neos/Ui/Aspects/PolicyAspect.php

The next thing is children rendering, which I removed in the PR above.

Then generating `preivewUris` takes a significan't amount of time.

If you remove all of that you end up with `0.58s` to render 540 nodes, which is OK.

Obviously we need to start with doing something about the policy aspect...

Here are the measurements themselves:

|Measurement|time, s|
|---|---|
|Original|24|
|No AOP|7.3|
|No AOP, No children|3.6|
|No AOP, No children, omitMostPropertiesForTreeState|2.6|
|No AOP, No children, omitMostPropertiesForTreeState, noPreviewUri|0.58|

File: Classes/Neos/Neos/Ui/Fusion/Helper/NodeInfoHelper.php [hunk 1]
```diff
@@ -103,20 +105,6 @@ public function renderNode(NodeInterface $node, ControllerContext $controllerCon
                 $nodeInfo['previewUri'] = $this->uri($nodeInLiveWorkspace, $controllerContext);
             }
         }
-
-        // child nodes for document tree, respecting the `baseNodeType` filter
-        $documentChildNodes = $node->getChildNodes($baseNodeType);
```
Comment (skurfuerst) [inline Classes/Neos/Neos/Ui/Fusion/Helper/NodeInfoHelper.php]: @dimaip does the BaseNodeType filter in the tree still work? :-) Just asking

Comment (dimaip) [reply Classes/Neos/Neos/Ui/Fusion/Helper/NodeInfoHelper.php]: @skurfuerst yes, absolutely, just tested it with `baseNodeType: 'Neos.Neos:Document,!Neos.Neos:Shortcut'`:

image
`

**Comment by dimaip** [comment]
Found a few edge cases, will fix tonight.

**Comment by dimaip** [comment]
Duh, unfortunately I have to give up on this one. My PR was great, but it didn't consider one usecase: checking if nodes have children... And if I do extra request to check it, it would be not much different from how it works currently, performance-wise.
So I'm giving up in favor of #1640 

Anyways, the performance killer is checking for policies currently, before we don't fix that there's no sense in optimizing anything else...

**Pull request closed without merging**